### ISSUE TITLE
Internal Pandora error (Hermes 1.3.1)

### ISSUE THREAD

**Issue by tomlogic** [HermesApp/Hermes#368]
Just today Hermes started failing with "An error has occurred with Pandora", a "Try Again" button, and "Internal Pandora error" below that.  I was listening to my station and it just stopped at the end of a song.

I seem to recall needing to make changes a few years ago so Hermes would continue working (updating certificates).  Perhaps I need to do those magic things again?

**Comment by tomlogic** [comment]
Possibly related to #337 and #340, but the current certificates seem to be OK.  Safari accepts both certificates (for `tuner` and `internal-tuner`) as valid.  The certificates I had manually trusted using instructions in #337 expired back in 2019, so my system hasn't been using them.

I'm using macOS Catalina on an Intel MacBook Pro.

**Comment by khudson** [comment]
I'm experiencing the same thing - trying to locate the logs to see if i can glean some additional information. once I figure something out I'll post details and see what it would take to come up with a fix.

**Comment by theksmith** [comment]
Same problem here today, worked yesterday. Pandora.com does work OK when loaded in Chrome.

I open the app and get a spinner for a few seconds, then same error/button as mentioned by the OP. I'm using a paid Pandora account and am on MacOS 11.4. No updates/upgrades done in the past few days. 

Tried deleting the Hermes folder from Application Support, and also the keychain entry. Upon new launch i was prompted for my Pandora Login, but once I submit the form, problem persists.

from Console:

error	17:45:25.469091-0700	Hermes	cannot open file at line 44595 of [02c344acea]
error	17:45:25.469116-0700	Hermes	os_unix.c:44595: (2) open(/var/db/DetachedSignatures) - No such file or directory
error	17:45:25.620270-0700	Hermes	MTLIOAccelDevice bad MetalPluginClassName property (null)
error	17:45:25.620985-0700	Hermes	+[MTLIOAccelDevice registerDevices]: Zero Metal services found
error	17:45:25.758404-0700	Hermes	nw_endpoint_handler_set_adaptive_read_handler [C1.1 IPv4#376ebd05:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for read_timeout failed
error	17:45:25.758449-0700	Hermes	nw_endpoint_handler_set_adaptive_write_handler [C1.1 IPv4#376ebd05:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for write_timeout failed
error	17:45:26.054824-0700	Hermes	nw_endpoint_handler_set_adaptive_read_handler [C2.1 IPv4#376ebd05:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for read_timeout failed
error	17:45:26.054852-0700	Hermes	nw_endpoint_handler_set_adaptive_write_handler [C2.1 IPv4#376ebd05:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for write_timeout failed
error	17:45:26.336253-0700	Hermes	nw_endpoint_handler_set_adaptive_read_handler [C3.1 IPv4#ca9a73f3:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for read_timeout failed
error	17:45:26.336283-0700	Hermes	nw_endpoint_handler_set_adaptive_write_handler [C3.1 IPv4#ca9a73f3:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for write_timeout failed
error	17:45:26.617537-0700	Hermes	nw_endpoint_handler_set_adaptive_read_handler [C4.1 IPv4#ca9a73f3:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for read_timeout failed
error	17:45:26.617590-0700	Hermes	nw_endpoint_handler_set_adaptive_write_handler [C4.1 IPv4#ca9a73f3:443 ready socket-flow (satisfied (Path is satisfied), viable, interface: utun2, ipv4, dns)] unregister notification for write_timeout failed

**Comment by khudson** [comment]
In order to debug this i think i need to set up mitmproxy or something to bust the ssl and inspect the request, then funnel hermes through it and see what the actual request and response is.
then do the same thing with the official client and see where the two disagree. i just haven't had enough time to mess with it today.

**Comment by nriley** [comment]
Might be easier just to insert logging if you can get it to build. Either way thanks for investigating! My time to spend on Hermes continues to be 0 unfortunately. 

Nicholas

**Comment by khudson** [comment]
I'm sure i've gotten it to build as i'm running one i built myself - but I built it like a year and a half ago and I couldn't find the buildenv I used to do so, so it'll probably take a bit for me to get it back up and running again. it's likely faster (but probably not easier!) for me to get the mitmproxy set up, as painful as that sounds. i'll see what i can do, but it might not be until tomrorow.

**Comment by sandinak** [comment]
I too am trying to do a build to see wazzup ... will try to TS this weekend at latest.

**Comment by sandinak** [comment]
Well .. went to start the build this morning .. and now Hermes is again working?  Anyone else?

**Comment by tomlogic** [comment]
Still not working for me.

**Comment by arsinio** [comment]
Wasn't working for me yesterday, working for me today!

**Comment by tomlogic** [comment]
And now it's working for me.  Certainly makes it difficult to troubleshoot...

I'll leave this open for another week and then if someone pings me I'll close it out.

**Comment by nriley** [comment]
Works for me too. Seems like this may have been on the Pandora side. I know Hermes does a not-great job of exposing errors, so it may be useful still trying to work on better logging/display if anyone has time.

**Comment by khudson** [comment]
Yup, it started working here as well - someone on that side probably made a boo boo and it likely borked a bunch of integrations with other devices, if i had to guess (without having tested that at all). For now, I'm glad I don't have to go through the debugging exercise, since i don't really have the time to do it!

I just wish we could find someone that *did* have the time and experience to maintain Hermes. I'd offer, but I don't have any ObjC experience to speak of.

**Comment by JohnLockeNJ** [comment]
Got the same error. Trusting certificates for tuner and internal-tuner manually isn't helping. Ideas?

**Comment by JohnLockeNJ** [comment]
It randomly started working again for me. Not sure what happened.
